### ISSUE TITLE
Binding from module to Windows File System

### ISSUE THREAD

**Issue by Steve0212a** [Azure/iotedge-eflow#76]
I am running the IOT Edge for Windows 1.1 and am attempting to get persistent storage working.  It appears that IOT Edge for Windows 1.1 runs a Linux docker container that then runs the agent/hub/custom modules inside the linux docker container.  I have been unable to get persistent storage working and cannot find a guide on how to make it work such that the container has access to the Windows file system.

I have reviewed this document, but it does not get the module access to the Windows file system.  https://docs.microsoft.com/en-us/azure/iot-edge/how-to-access-host-storage-from-module?view=iotedge-2018-06.

I tried adding this to the container create options, but it did not work.
```
{
  "HostConfig": {
    "Binds": [
      "c:\\temp\\Edge\\IotEdgeModule1:/tmp/IotEdgeModule1"
    ]
  }
}
```

I got this error:
```
Sep 17 14:44:46 BLAZIN-EFLOW iotedged[6562]: 2021-09-17T14:44:46Z [WARN] - Could not create module iotedgemodule1005
Sep 17 14:44:46 BLAZIN-EFLOW iotedged[6562]: 2021-09-17T14:44:46Z [WARN] -         caused by: invalid mode: /tmp/IotEdgeModule1
```

Can you please provide the details on how to make this work?

**Comment by TerryWarwick** [comment]
@Steve0212a,

We currently do not support file/folder sharing between the Linux VM and host, but it is something that we are considering to offer in the future.  I believe that binding approach would only work inside the VM.

Could I get you to contact me directly at Terry.Warwick@microsoft.com and share whatever details of your scenario that you can with us so that we can take this into consideration as we are exploring the feasibility of enabling this functionality.

Terry Warwick
Microsoft

**Comment by fcabrera23** [comment]
Hi,

We currently do not support file/folder sharing between the Linux VM and host. For certificate management for IoT Edge, the approach would be the following:

1. Deploy EFLOW
2. Copy the certificates from the Windows host to the EFLOW VM using the [Copy-EflowVmFile](https://docs.microsoft.com/en-us/azure/iot-edge/reference-iot-edge-for-linux-on-windows-functions?view=iotedge-2018-06#copy-eflowvmfile)
3. Provision EFLOW
4. [Manage certificates on an IoT Edge device]https://docs.microsoft.com/en-us/azure/iot-edge/how-to-manage-device-certificates?view=iotedge-2020-11&tabs=linux

Thanks,
Francisco

**Comment by scout208** [comment]
Hi, sorry to bring this up on a closed thread, but this is a pretty critical feature for us to begin using EFLOW. Currently we are using Windows containers because they support this but we would really like to switch to EFLOW. Is there any chance that this feature would be supported in a future version of EFLOW and if so an estimated timeline on that? I am willing to share more specific details of our scenario in case it would help or if there is any potential workaround. Thanks!

**Comment by bkarrlander** [comment]
> @Steve0212a,
> 
> We currently do not support file/folder sharing between the Linux VM and host, but it is something that we are considering to offer in the future. I believe that binding approach would only work inside the VM.
> 
> Could I get you to contact me directly at [Terry.Warwick@microsoft.com](mailto:Terry.Warwick@microsoft.com) and share whatever details of your scenario that you can with us so that we can take this into consideration as we are exploring the feasibility of enabling this functionality.
> 
> Terry Warwick Microsoft

Hi, @TerryWarwick , @fcabrera23. Have you explored the possibility of sharing file/folder between the VM and host? 

Background: On our pure Linux devices we are collecting files from shared folders on Windows machines. On some sites we currently can't install separate Linux-devices and have to use EFLOW in these cases.

**Comment by fcabrera23** [comment]
Hi @bkarrlander,

We are currently working on the file/folder sharing feature, and we expect to have it for our June update, although final date still TBD.

Thanks,
Francisco

**Comment by bkarrlander** [comment]
Thank you, @fcabrera23. Sounds great. 

Regards,
Björn

**Comment by andrewhickman** [comment]
> Hi @bkarrlander,
> 
> We are currently working on the file/folder sharing feature, and we expect to have it for our June update, although final date still TBD.
> 
> Thanks, Francisco

Hi @fcabrera23, do you have any update on the status of this feature?

**Comment by fcabrera23** [comment]
Hi @andrewhickman,

We plan to release the feature in our next EFLOW CR release. This update will contain considerable improvements (Folder sharing, Mariner2.0 upgrade, IoT Edge 1.3.0 upgrade), so we are making during that everything works fine. Probably will be released in 1-2 weeks. 

Thanks,
Francisco

**Comment by Prov-Matthias** [comment]
Hi @fcabrera23,

any news for that? We have the same problem... 

We use it for differnt topics, for example we have etablished a deployment process (Bamboo pipline) for the published.nodes.json. And that is just one use case.

But I am strugle to draw the new migration concept, without the folder sharing.
And in the Securitiy review, everyone is very precise when it comes to such topics ;)

Thanks,
Matthias

**Comment by fcabrera23** [comment]
Hi @Prov-Matthias,

We are releasing the Folder Sharing mechanism in our next EFLOW CR update, probably this week. This will only be supported for Windows 11 for the months while we work on the backport to Win10. 

Thanks,
Francisco

**Comment by Prov-Matthias** [comment]
Hi @fcabrera23,

Sounds very good :) Can I then assume that Win11 also means Windows Server 2022?

Thanks,
Matthias

**Comment by fcabrera23** [comment]
Hi @Prov-Matthias,

Windows Server 2019/2022 will be considered in the future. Currently, we will only support Win10/11, although we are open to discussing it with customers.

Thanks,
Francisco

**Comment by maggiojoe2** [comment]
Are there any updates on the Windows 10 version of this feature?

**Comment by fcabrera23** [comment]
Hi @maggiojoe2,

Thanks for reaching out. We had a Windows dependency that was backported and released with 11C November update. This backporting will enable Folder sharing with Win10 (21H2), Win11, and Win Server 2019 and 2022.
However, we started some tests with these new versions last week, and we found some bugs. We expect to investigate these bugs this week and will have more clarity on an ETA for these versions.

Thanks,
Francisco
